### ISSUE TITLE
Unable to detech hash salt and rounds in minimal docker install

### ISSUE THREAD

**Issue by Svarto** [clems4ever/authelia#307]
Hi,

I tried to set up a test environment with Authelia, to try it out before committing it to my homelab. I have set it up and can correctly navigate to the webpage, and get the login page, on http://home-ip:8080.

However, when I try to login, using a different hashed password than "mypassword", I get an authentication failure and the following error in my logs:

`error: date='Sun Dec 16 2018 20:47:29 GMT+0000 (UTC)' method='POST', path='/api/firstfactor' requestId='xxx' sessionId='xxx' ip='desktop-ip' message='Reply with error 200: Unable to detect the hash salt and rounds. Make sure the password is hashed with SSHA512.'`

So it can find the file, but somehow not detect if the hash is correct. I have the following in my users_database.yml:

```
###############################################################
#                         Users Database                      #
###############################################################

# This file can be used if you do not have an LDAP set up.

# List of users
users:
  test_user:
    password: "{CRYPT}$6$rounds=500000$W/8aYO.7Nb2SqIgd$TOXDeaMicSWZ.9BU/VafTxjVICwFl77zvR2NT0AV/tJ7pscfEgr7omY/CRN.Gdqbr7XKb6DsKnHbWNrRSFgtm0"
    email: myemail@mail.com
    groups:
      - admins
      - dev
```

Hope you can help me?

TLDR: Error is thrown when using a different password to the default mypassword.

**Comment by clems4ever** [comment]
Hello @Svarto , what method to you use to generate the hash? Do you follow the doc?

**Comment by Svarto** [comment]
Yes, I literally copy and paste the code from the docs using python3 etc. I use python version 3.7. The weird thing is that when I leave it as "mypassword" it works fine, but when I change that to "testtest" I get the error that the hash can't be read.

Am I assuming correctly creating the hash for another password I just change the "mypassword" to my own password string within the " " and change nothing else in the command string?

-------- Original Message --------

**Comment by clems4ever** [comment]
@Svarto , you're doing the right thing. I will test on my side to see if I reproduce. I'll let you know.
In the meantime what you can do to generate a valid hash is to use the reset password feature. It will update the conf with the hash of the new password and this one should work.

**Comment by Svarto** [comment]
it worked with the password reset, I will close this now. Thanks a lot for the help!

**Comment by Roemer** [comment]
I have the very same problem.  I also cannot even use the password reset as it somehow returns `{"error":"An error occurred during password reset. Your password has not been changed."}`

**Comment by clems4ever** [comment]
Hello @Roemer, do you have more info from the logs?
Anyway, I will release a new major in a few days, I will include a tool to generate a hash given a password. Stay tuned.
